plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlClient.java

```diff
@@ -377,6 +387,44 @@ public void testConvertNotExpression()
                 .hasValue("NOT ((\"c_varchar\") IS NOT NULL)");
     }
 
+    @Test(dataProvider = "testTypes")
+    public void testConvertCast(Type testedType)
+    {
+        assertThat(JDBC_CLIENT.convertPredicate(SESSION,
+                translateToConnectorExpression(
+                        new Cast(
+                                new SymbolReference("value_symbol"),
+                                toSqlType(testedType)),
+                        Map.of("value_symbol", testedType)),
+                Map.of("value_symbol", VARCHAR_COLUMN)))
+                .hasValue(format("CAST (\"c_varchar\" AS %s)", JDBC_CLIENT.mapTypeSignature(SESSION, testedType.getTypeId()).get()));
+    }
+
+    @Test
+    public void testConvertCastWithUnsupportedType()
+    {
+        assertThat(JDBC_CLIENT.convertPredicate(SESSION,
+                translateToConnectorExpression(
+                        new Cast(
+                                new SymbolReference("value_symbol"),
+                                toSqlType(ROW_TYPE)),
+                        Map.of("value_symbol", ROW_TYPE)),
+                Map.of("value_symbol", VARCHAR_COLUMN)))
+                .isEmpty();
+    }
+
+    @DataProvider
+    public Object[][] testTypes()
+    {
+        return new Object[][]{
+                {createVarcharType(1337)},
+                {BIGINT},
+                {TIMESTAMP_TZ_SECONDS},
+                {TIMESTAMP_TZ_NANOS},
```

Comment:
PostgreSQL does not have equivalent of TIMESTAMP_TZ_NANOS, so pushing down a CAST to such type into PostgreSQL isn't correct